tests/e2e/privacy.spec.js

```diff
@@ -56,50 +63,143 @@ describe('Privacy', () => {
     });
   });
 
-  describe('Ad-Blocking', () => {
+  describe.only('Ad-Blocking', () => {
     beforeEach(() => updatePrivacySettings('ad-blocking', false));
 
-    const WEBSITE_URL = 'https://www.onet.pl/';
-    const SELECTOR = 'div[class^="AdSlotPlaceholder_"]';
-
-    it('displays ads on a page', async () => {
-      await browser.url(WEBSITE_URL);
+    const SELECTOR = 'ad-slot';
 
-      let displayed = false;
-      for (const ad of await $$(SELECTOR)) {
-        if (displayed) break;
-        displayed = await ad.isDisplayed();
-      }
-
-      expect(displayed).toBe(true);
+    it('does not block ads on a page', async () => {
+      await browser.url(PAGE_URL);
+      await expect($(SELECTOR)).toBeDisplayed();
     });
 
     it('blocks ads on a page', async () => {
       await updatePrivacySettings('ad-blocking', true);
 
+      await browser.url(PAGE_URL);
+      await expect($(SELECTOR)).not.toBeDisplayed();
+    });
+  });
+
+  describe('Anti-Tracking', () => {
+    beforeEach(() => updatePrivacySettings('anti-tracking', false));
+
+    const TRACKER_IDS = ['facebook_connect', 'pinterest_conversion_tracker'];
+
+    it('does not block tracker requests on the page', async () => {
+      await disableCache();
+      await browser.url(PAGE_URL);
+
+      await switchToPanel(async () => {
```

Comment:
how about a different syntax:
```js
const goBack = await switchToPanel();

// regular tests

await goBack();
```

Reply:
Switching can throw errors, so you can hang in some state, where the focus could still be on another window. 

This pattern ensures, that whatever happens, after `await switchToPanel(...)` you are back to the main context. Your pattern does not guarantee that.

Reply:
if it hangs then it should timeout, if something else goes wrong, it should throw.

not sure if I see the benefit of nested syntax

Reply:
I would have to show you an example, but working with real browser is not that easy. In your pattern, there is no explicit clean up, if something goes wrong between switch and go back. This is required, as the browser may stay focused on the incorrect window. Because of that, the whole code must be in try/catch call, and a clean up of the switch must be made if something throws an error.